publisher/sql/sanitize.sql

```diff
@@ -1,37 +1,92 @@
-UPDATE emails
-SET
-  email            = regexp_replace(email, '(^[^@]+)', md5(email)),
-  "recipientEmail" = regexp_replace("recipientEmail", '(^[^@]+)', md5("recipientEmail")),
-  name             = md5(name);
-
-UPDATE emails
-SET
-  phone = 93000000
-WHERE phone IS NOT NULL;
-
-UPDATE users
-SET
-  email         = regexp_replace(email, '(^[^@]+)', md5(email)),
-  "displayName" = md5("displayName");
-
-UPDATE users
-SET
-  phone = 93000000
-WHERE phone IS NOT NULL;
-
-UPDATE contacts
-SET
-  phone = '93000000'
-WHERE phone IS NOT NULL;
-
-UPDATE contacts
-SET
-  name  = md5(name),
-  email = regexp_replace(email, '(^[^@]+)', md5(email)),
-  "fromEmail" = regexp_replace("fromEmail", '(^[^@]+)', md5("fromEmail"));
-
-UPDATE members
-SET
-  "firstName" = RIGHT(md5("firstName"), 20),
-  "lastName"  = RIGHT(md5("lastName"), 20),
-  email       = RIGHT(regexp_replace(email, '(^[^@]+)', md5(email)), 40);
+
+-- We define sanitization functions first:
+-- - sanitize_email
+-- - sanitize_phone
+-- - sanitize_value
+CREATE OR REPLACE FUNCTION sanitize_email(_t VARCHAR, _c VARCHAR)
+  RETURNS VOID AS
+$$
+DECLARE
+  affected NUMERIC;
+BEGIN
+  EXECUTE format('UPDATE "%s" SET "%s" = RIGHT(md5("%s"), 12) || SUBSTRING("%s", ''@.+$'');', _t, _c, _c, _c);
+  GET DIAGNOSTICS affected = ROW_COUNT;
+  RAISE NOTICE '%.% is sanitized: % rows affected', _t, _c, affected;
+END
+$$
+LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION sanitize_phone(_t VARCHAR, _c VARCHAR)
+  RETURNS VOID AS
+$$
+DECLARE
+  affected NUMERIC;
+BEGIN
+  EXECUTE format('UPDATE "%s" SET "%s" = ''93000000'' WHERE "%s" IS NOT NULL;', _t, _c, _c);
+  GET DIAGNOSTICS affected = ROW_COUNT;
+  RAISE NOTICE '%.% is sanitized: % rows affected', _t, _c, affected;
+END
+$$
+LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION sanitize_value(_t VARCHAR, _c VARCHAR)
+  RETURNS VOID AS
+$$
+DECLARE
+  affected NUMERIC;
+BEGIN
+  EXECUTE format('UPDATE "%s" SET "%s" = RIGHT(md5("%s"), 12);', _t, _c, _c);
+  GET DIAGNOSTICS affected = ROW_COUNT;
+  RAISE NOTICE '%.% is sanitized: % rows affected', _t, _c, affected;
+END
+$$
+LANGUAGE plpgsql;
+
+-- This query is for demo purposes only. It shows all the columns with non-null comments.
+-- The query is a base for filtering columns to sanitize.
```

Comment:
let's comment it out?

Reply:
We can, but I think output like this is useful:
```
publisher_1_4a974b09595b |  table_name |  column_name   |    description    
publisher_1_4a974b09595b | ------------+----------------+-------------------
publisher_1_4a974b09595b |  contacts   | name           | SANITIZE_AS_VALUE
publisher_1_4a974b09595b |  contacts   | email          | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  contacts   | phone          | SANITIZE_AS_PHONE
publisher_1_4a974b09595b |  contacts   | recipientEmail | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  contacts   | fromEmail      | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  emails     | name           | SANITIZE_AS_VALUE
publisher_1_4a974b09595b |  emails     | email          | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  emails     | phone          | SANITIZE_AS_PHONE
publisher_1_4a974b09595b |  emails     | recipientEmail | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  members    | firstName      | SANITIZE_AS_VALUE
publisher_1_4a974b09595b |  members    | lastName       | SANITIZE_AS_VALUE
publisher_1_4a974b09595b |  members    | phone          | SANITIZE_AS_PHONE
publisher_1_4a974b09595b |  members    | phone2         | SANITIZE_AS_PHONE
publisher_1_4a974b09595b |  members    | email          | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  users      | email          | SANITIZE_AS_EMAIL
publisher_1_4a974b09595b |  users      | phone          | SANITIZE_AS_PHONE
publisher_1_4a974b09595b |  users      | displayName    | SANITIZE_AS_VALUE
```

Reply:
agree, let's preserve it